src/main/java/com/epam/test_generator/services/JiraService.java

```diff
@@ -0,0 +1,332 @@
+        caze.setJiraParentKey(jiraSubTask.getJiraParentKey());
+        caze.setJiraProjectKey(jiraSubTask.getJiraProjectKey());
+        caze.setLastModifiedDate(LocalDateTime.now());
+        caze.setLastJiraSyncDate(caze.getLastModifiedDate());
+
+        caseDAO.save(caze);
+
+        Suit suit = suitDAO.findByJiraKey(jiraSubTask.getJiraParentKey());
+        checkNotNull(suit);
+        suit.getCases().add(caze);
+    }
+
+
+    /**
+     * Updates existent suit in BDD by related Jira story.
+     *
+     * @param jiraStory - related issue from Jira
+     */
+    private void updateSuitFromJiraStory(JiraStory jiraStory) {
+        Suit suit = suitDAO.findByJiraKey(jiraStory.getJiraKey());
+        if (suit == null) {
+            return;
+        }
+        suit.setName(jiraStory.getName());
+        suit.setDescription(jiraStory.getDescription());
+        suit.setPriority(getPriority(jiraStory.getPriority()));
+        suit.setLastJiraSyncDate(LocalDateTime.now());
+
+        suitDAO.save(suit);
+    }
+
+
+    /**
+     * Updates existent case in BDD by related Jira sub story.
+     *
+     * @param jiraSubTask - related issue from Jira
+     */
+    private void updateSubTaskFromJiraSubTask(JiraSubTask jiraSubTask) {
+        Case caze = caseDAO.findByJiraKey(jiraSubTask.getJiraKey());
+        if (caze == null) {
+            return;
+        }
+        caze.setName(jiraSubTask.getName());
+        caze.setDescription(jiraSubTask.getDescription());
+        caze.setPriority(getPriority(jiraSubTask.getPriority()));
+        caze.setLastJiraSyncDate(LocalDateTime.now());
+        caseDAO.save(caze);
+    }
+
+    private Integer getPriority(String priority) {
+        if (priority.equals("Critical")) {
+            return 1;
+        }
+        if (priority.equals("Major")) {
+            return 2;
+        }
+        if (priority.equals("Minor")) {
+            return 3;
+        }
+        if (priority.equals("Trivial")) {
+            return 4;
+        }
+        if (priority.equals("Blocker")) {
+            return 5;
+        }
+        return null;
+    }
+
+
+    /**
+     * Returns all projects in Jira but non-existent in BBD
+     *
+     * @return list projects
+     */
+    public List<JiraProject> getNonexistentJiraProjects() throws JiraException {
+        return jiraProjectDAO.getAllProjects().stream()
+                .filter(project -> projectDAO.findByJiraKey(project.getJiraKey()) == null)
+                .collect(Collectors.toList());
+    }
+
+
+    /**
+     * Synchronisation method from Jira
+     * This method receives all related and opened stories and sub stories from Jira and update its in BDD.
+     * <p>
+     * In case where issue has closed in Jira - the changes not present in the response. By this reason
+     * all existing cases or suits which not presents in response will be deleted from BBD.
+     */
+    public void syncFromJiraToBDD() throws JiraException {
```

Comment:
It is a huge method. I looked through and think that it has three parts. First when we work with jiraStories, second part is jiraSubTasks and the last part is suits and cases. So maybe extract these parts to another private methods.